plugin-api/src/main/java/org/sonar/api/batch/sensor/issue/NewExternalIssue.java

```diff
@@ -53,19 +55,37 @@ public interface NewExternalIssue {
 
   /**
    * Type of issue.
+   * @deprecated since 10.1, use {@link #addImpact(SoftwareQuality softwareQuality, org.sonar.api.issue.impact.Severity severity)()} instead
    */
+  @Deprecated(since = "10.1")
   NewExternalIssue type(RuleType type);
 
+  /**
+   * Attribute of the issue according to Clean Code Taxonomy.
+   * Providing it is optional for now but will become mandatory in the future.
+   * @since 10.1
+   */
+  NewExternalIssue cleanCodeAttribute(CleanCodeAttribute attribute);
```

Comment:
Clarification question for me - will the behaviour for `ExternalIssue` be different than for 'normal' `Issue`?  The issues get attributes from the rule, they won't have attributes on their own. Won't it be the case here?

Reply:
Would be nice to document the answer if this was answered in other channels